src/integrationTest/java/com/mongodb/hibernate/id/MongoIdFieldNameIntegrationTests.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.id;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.hibernate.internal.cfg.MongoConfigurationBuilder;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import org.bson.BsonDocument;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.SessionFactoryScopeAware;
+import org.junit.jupiter.api.AutoClose;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+@SessionFactory(exportSchema = false)
+@DomainModel(
+        annotatedClasses = {
+            MongoIdFieldNameIntegrationTests.EntityWithoutIdColumnAnnotation.class,
+            MongoIdFieldNameIntegrationTests.EntityWithIdColumnAnnotationWithoutNameElement.class,
+            MongoIdFieldNameIntegrationTests.EntityWithIdColumnAnnotationWithValidNameElement.class
+        })
+class MongoIdFieldNameIntegrationTests implements SessionFactoryScopeAware {
+
+    @AutoClose
+    private MongoClient mongoClient;
+
+    private MongoCollection<BsonDocument> collection;
+
+    private SessionFactoryScope sessionFactoryScope;
+
+    @Override
+    public void injectSessionFactoryScope(SessionFactoryScope sessionFactoryScope) {
+        this.sessionFactoryScope = sessionFactoryScope;
+    }
+
+    @BeforeAll
+    void beforeAll() {
+        var config = new MongoConfigurationBuilder(
+                        sessionFactoryScope.getSessionFactory().getProperties())
+                .build();
+        mongoClient = MongoClients.create(config.mongoClientSettings());
+        collection = mongoClient.getDatabase(config.databaseName()).getCollection("movies", BsonDocument.class);
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        collection.drop();
+    }
+
+    @Test
```

Comment:
I'm unclear why there are no assertions in these tests.  Is it the case that the only thing we need to test is that no exception is thrown?  I was expecting that we would assert that the inserted document has the expected value for `_id` (in this particular test, the value of `1`).

(Note that MongoDB driver behavior is that, if a document is inserted without an `_id` field, the driver will generate a random `Objectid` for the `_id`).  So if we have a bug in Hibernate dialect you could end up with a document like

```
{ 
   _id: <random ObjectId>,
   id: 1
}
```

Reply:
yeah, my original rationale is based on the assumption that without `_id` field, insertion would fail.
Added verification logic based on the stored doc content.